native/src/main/java/io/ballerina/lib/confluent/CustomSchemaRegistryClient.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.confluent;
+
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import org.apache.avro.Schema;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.ballerina.lib.confluent.ModuleUtils.BASE_URL;
+import static io.ballerina.lib.confluent.ModuleUtils.HEADERS;
+import static io.ballerina.lib.confluent.ModuleUtils.IDENTITY_MAP_CAPACITY;
+import static io.ballerina.lib.confluent.ModuleUtils.ORIGINALS;
+
+public class CustomSchemaRegistryClient {
+
+    SchemaRegistryClient schemaRegistryClient;
+
+    public CustomSchemaRegistryClient(BMap<?, ?> config) {
+        BString baseUrl = (BString) config.get(BASE_URL);
+        long identityMapCapacity = config.getIntValue(IDENTITY_MAP_CAPACITY);
+        BMap<?, ?> originals = config.getMapValue(ORIGINALS);
+        BMap<?, ?> httpHeaders = config.getMapValue(HEADERS);
```

Comment:
Better to use the exact types here. From the Ballerina type, we can ensure the type of the maps.